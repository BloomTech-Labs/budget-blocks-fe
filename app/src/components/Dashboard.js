import React,{useEffect,useState} from 'react';
import { connect } from "react-redux";
import styled from 'styled-components'
import Bootstrap from 'bootstrap/dist/css/bootstrap.min.css';
import {Row,Col} from 'react-bootstrap'
import { Progress,Container,Table } from "reactstrap";
import './table.css';
import LinkedComponent from "./Balance/LinkedComponent";
import UnlinkedComponent from "./Balance/UnlinkedComponent";
function Dashboard(props) {
 const Div = styled.div`
background-color:lightgrey;
padding:25px;
`
const GreenButton = styled.button`
width: 2.5rem; 
height: 2.5rem;
border-radius: 50%;
background-color:lightgreen;
border-color:lightgreen;
font-size:1.2rem;
color:white;
margin-left:2rem;
`
const RedButton = styled.button`
width: 2.5rem; 
height: 2.5rem;
border-radius: 50%;
background-color:red;
border-color:red;
font-size:1.2rem;
color:white;
margin-left:2rem;
` 

function AddBalance(){
  if(props.LinkedAccount){
    return(<LinkedComponent/>)
  }else{
    return(<UnlinkedComponent/>)
  }
}
  return (
   
   <Container>
  <Row >
    <Col sm={9}> <div>
    <Div>
      <Row>
      <Col sm={8}> <h1>Hi, First Name </h1><p>Ksharper@studentmba.org</p> </Col>
      <Col sm={4}>KS</Col>
      </Row>   
  </Div>
  <Table responsive>
  <thead>
    <tr>
      <th></th>
      <th>Block</th>
      <th>Current status</th>
      <th></th>
      <th>Total Expenses</th>
      <th>Limit</th>
      <th>Action</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td></td>
      <td>Title</td>
      <td>
          <Progress value={90} />
     </td>
      <td>&#9989;</td>
      <td>$0.00</td>
      <td>$500.00</td>
      <td>Edit</td>
    </tr>
    <tr>
      <td></td>
      <td>Title</td>
      <td>
          <Progress value={90} /></td>
      <td>&#9989;</td>
      <td>$0.00</td>
      <td>$500.00</td>
      <td>Edit</td>
    </tr>
    <tr>
      <td></td>
      <td>Title</td>
      <td>
          <Progress value={90} /></td>
      <td>&#9989;</td>
      <td>$0.00</td>
      <td>$500.00</td>
      <td>Edit</td>
    </tr>
  </tbody>
</Table>
  </div></Col>
    <Col sm={3}> <div>
      <Div>
    <Container>
        <div className="center">
          <h3>Total Budget</h3>
          <h3>$200.00</h3>
          <Progress value={90} />
        </div>
      </Container>
  <div className="buttons">
 
    <div className="green-button">
    <Row>
      <Col sm={6}><b>+$1250.67</b><h5>Income</h5><GreenButton>+</GreenButton></Col><Col sm={6}><b>-$567.56</b><h5>Expenses</h5><RedButton>-</RedButton></Col>
    </Row>
    {/* <Row>
      <Col sm={6}><h5>Income</h5></Col><Col sm={6}><h5>Expenses</h5></Col>
    </Row> */}
   
   
  </div>
 
      </div>
      </Div>
      <div className="subheading">
      <Row>
        {AddBalance()}
      </Row>
      </div>
      <div>
      <div className="subheading">
      <Row>
        <Col sm={9}>Upcoming Due Dates </Col>
      </Row>
      <Row>
        <Col sm={7}><p>Rent</p></Col><Col sm={3}><p>-$760.00</p><p>03/01/2020</p></Col>
      </Row>
      </div>
      </div>
</div></Col>
  </Row>
  
  <Row>
    <Col sm={9}> 
    <h2>Recent Transactions</h2>
    <Div className="border">
    <Row>
      <Col sm={9}>Purchase Authorized on 01/01 Storage Comapany for User </Col>
      
    </Row>
    <Row>
      <Col sm={9}><p>01/01/2020</p></Col><Col sm={3}><p>-$50.00</p></Col>
    </Row>
    <Row>
      <Col sm={9}><p>Category</p></Col><Col sm={3}><p>Utilities</p></Col>
    </Row>
    </Div>
    <Div className="border">
    <Row>
      <Col sm={9}>Purchase Authorized on 01/01 Storage Comapany for User </Col>
      
    </Row>
    <Row>
      <Col sm={9}><p>01/01/2020</p></Col><Col sm={3}><p>-$50.00</p></Col>
    </Row>
    <Row>
      <Col sm={9}><p>Category</p></Col><Col sm={3}><p>Utilities</p></Col>
    </Row>
    </Div>
    </Col>
    <Col sm={3}><div className="subheading">  <Row>
      <Col sm={10}><p>Savings Goals</p></Col>
    </Row>
    <Row>
      <Col sm={7}><p>Emergency Funds</p></Col><Col sm={3}><p>$2750.00</p><p>01/01/2020</p></Col>
    </Row>
    </div>
    </Col>
  </Row>
</Container> 
  
  );
}

function mapStateToProps(state){
  return {
      LinkedAccount: state.loginReducer.user.LinkedAccount
  }
}

export default connect(mapStateToProps,{})(Dashboard)