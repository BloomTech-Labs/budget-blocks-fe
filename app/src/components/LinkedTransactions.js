import React from 'react';
import { makeStyles, useTheme } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import AddIcon from '@material-ui/icons/Add';
import { getTransactions } from "../redux/actions/PlaidActions";
import { connect } from "react-redux";
// Generate Order Data
function createData(id, date, description, category, paymentMethod, amount) {
	return { id, date, description, category, paymentMethod, amount };
}

const useStyles = makeStyles(theme => ({
	root: {
		'& > *': {
			margin: theme.spacing(1)
		}
	},
	topcontent: {
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'space-between'
	},
	card: {
		display: 'flex',
		border: 'black',
		flexDirection: 'row',
		margin: '1%',
		background: 'lightgrey',
		justifyContent: 'space-between'
	},
	details: {
		display: 'flex'
	},
	content: {
		flex: '1 0 auto',
		alignItems: 'flex-start',
		textAlign: 'left',
		paddingLeft: theme.spacing(5)
	},
	controls: {
		display: 'flex',
		alignItems: 'center',
		textAlign: 'right',
		alignContent: 'flex-end',
		paddingRight: theme.spacing(5)
	},
	text: {
		textAlign: 'left',
		paddingLeft: theme.spacing(5)
	}
}));

//dummy data
const rows = [
	createData(
		0,
		'Jan 20, 2020',
		'Puchase Authorized on 01/01 Storage Company For USER 6A7BC81524XY1524',
		'Grocery',
		'VISA ⠀•••• 1234',
		312.44
	),
	createData(
		1,
		'Jan 21, 2020',
		'LA fitness',
		'Utility',
		'VISA ⠀•••• 1234',
		866.99
	),
	createData(
		2,
		'Jan 22, 2020',
		'APPPLE.COM/BILL',
		'Entertainment',
		'VISA ⠀•••• 1234',
		100.81
	),
	createData(
		3,
		'Jan 23, 2020',
		'Target',
		'Shopping',
		'Visa ⠀•••• 1234',
		654.39
	),
	createData(
		4,
		'Jan 24, 2020',
		'Nordstrom #0123',
		'Shopping',
		'VISA ⠀•••• 1234',
		212.79
	)
];

function preventDefault(event) {
	event.preventDefault();
}

const  Transactions = props =>  {
	const classes = useStyles();
	const theme = useTheme();
	
	let transactions = []
	props.transactions.map(i => i.transactions.map(i => transactions.push(i)))
	console.log(transactions)
	transactions = transactions.sort(function(a,b){
	  return new Date(b.payment_date) - new Date(a.payment_date)
	})
	return (
		<React.Fragment >
			<div className={classes.topcontent}>
				<b className={classes.text}>Recent Transactions:</b>
				<div className={classes.root}>
					<Button variant='contained'>
						<AddIcon />
						Add Transactions
					</Button>
					<Button variant='contained' color='primary'>
						<AddIcon />
						Add Income
					</Button>
				</div>
			</div>

			{transactions.map(i => (
				<Card className={classes.card} key={i.id}>
					<div className={classes.details}>
						<CardContent className={classes.content}>
							<Typography component='p' variant='body1'>
								{i.name}
							</Typography>
							<Typography variant='subtitle1' color='textSecondary'>
								{i.payment_date}
							</Typography>
							
						</CardContent>
					</div>
					<div className={classes.controls}>${Math.round(10 * i.amount)/100}</div>
				</Card>
			))}
		</React.Fragment>
	);
}
function mapStateToProps(state){
	return {
		userID:state.loginReducer.user.id,
		LinkedAccount:state.loginReducer.user.LinkedAccount,
		transactions:state.plaidReducer.categories.filter(i => i.transactions.length > 0)
	}
  }
  
  export default connect(mapStateToProps,{ getTransactions })(Transactions)

// export const Tran = props => { }
//const mapStateToProps = state =>{return{ }}
//export default connect(mapStateToProps, {}(trans  ))
