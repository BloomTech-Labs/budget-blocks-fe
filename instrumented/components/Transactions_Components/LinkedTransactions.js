var cov_1x4u3utbdj = function () {
  var path = "/Users/Cha/Desktop/Lambda/budget-blocks-fe/src/components/Transactions_Components/LinkedTransactions.js";
  var hash = "00b494f305c68c531cf84782caae147fd4c42eac";
  var global = new Function("return this")();
  var gcv = "__coverage__";
  var coverageData = {
    path: "/Users/Cha/Desktop/Lambda/budget-blocks-fe/src/components/Transactions_Components/LinkedTransactions.js",
    statementMap: {
      "0": {
        start: {
          line: 10,
          column: 18
        },
        end: {
          line: 59,
          column: 3
        }
      },
      "1": {
        start: {
          line: 10,
          column: 39
        },
        end: {
          line: 59,
          column: 1
        }
      },
      "2": {
        start: {
          line: 61,
          column: 21
        },
        end: {
          line: 101,
          column: 1
        }
      },
      "3": {
        start: {
          line: 66,
          column: 17
        },
        end: {
          line: 66,
          column: 28
        }
      },
      "4": {
        start: {
          line: 67,
          column: 29
        },
        end: {
          line: 67,
          column: 43
        }
      },
      "5": {
        start: {
          line: 68,
          column: 21
        },
        end: {
          line: 70,
          column: 2
        }
      },
      "6": {
        start: {
          line: 69,
          column: 2
        },
        end: {
          line: 69,
          column: 21
        }
      },
      "7": {
        start: {
          line: 72,
          column: 16
        },
        end: {
          line: 72,
          column: 46
        }
      },
      "8": {
        start: {
          line: 73,
          column: 25
        },
        end: {
          line: 73,
          column: 40
        }
      },
      "9": {
        start: {
          line: 74,
          column: 25
        },
        end: {
          line: 76,
          column: 2
        }
      },
      "10": {
        start: {
          line: 75,
          column: 2
        },
        end: {
          line: 75,
          column: 16
        }
      },
      "11": {
        start: {
          line: 77,
          column: 21
        },
        end: {
          line: 79,
          column: 2
        }
      },
      "12": {
        start: {
          line: 78,
          column: 2
        },
        end: {
          line: 78,
          column: 17
        }
      },
      "13": {
        start: {
          line: 81,
          column: 1
        },
        end: {
          line: 100,
          column: 2
        }
      },
      "14": {
        start: {
          line: 103,
          column: 1
        },
        end: {
          line: 107,
          column: 3
        }
      }
    },
    fnMap: {
      "0": {
        name: "(anonymous_0)",
        decl: {
          start: {
            line: 10,
            column: 29
          },
          end: {
            line: 10,
            column: 30
          }
        },
        loc: {
          start: {
            line: 10,
            column: 39
          },
          end: {
            line: 59,
            column: 1
          }
        },
        line: 10
      },
      "1": {
        name: "(anonymous_1)",
        decl: {
          start: {
            line: 61,
            column: 21
          },
          end: {
            line: 61,
            column: 22
          }
        },
        loc: {
          start: {
            line: 61,
            column: 30
          },
          end: {
            line: 101,
            column: 1
          }
        },
        line: 61
      },
      "2": {
        name: "(anonymous_2)",
        decl: {
          start: {
            line: 68,
            column: 21
          },
          end: {
            line: 68,
            column: 22
          }
        },
        loc: {
          start: {
            line: 68,
            column: 26
          },
          end: {
            line: 70,
            column: 2
          }
        },
        line: 68
      },
      "3": {
        name: "(anonymous_3)",
        decl: {
          start: {
            line: 74,
            column: 25
          },
          end: {
            line: 74,
            column: 26
          }
        },
        loc: {
          start: {
            line: 74,
            column: 31
          },
          end: {
            line: 76,
            column: 2
          }
        },
        line: 74
      },
      "4": {
        name: "(anonymous_4)",
        decl: {
          start: {
            line: 77,
            column: 21
          },
          end: {
            line: 77,
            column: 22
          }
        },
        loc: {
          start: {
            line: 77,
            column: 27
          },
          end: {
            line: 79,
            column: 2
          }
        },
        line: 77
      },
      "5": {
        name: "mapStateToProps",
        decl: {
          start: {
            line: 102,
            column: 9
          },
          end: {
            line: 102,
            column: 24
          }
        },
        loc: {
          start: {
            line: 102,
            column: 32
          },
          end: {
            line: 108,
            column: 1
          }
        },
        line: 102
      }
    },
    branchMap: {
      "0": {
        loc: {
          start: {
            line: 92,
            column: 24
          },
          end: {
            line: 92,
            column: 62
          }
        },
        type: "cond-expr",
        locations: [{
          start: {
            line: 92,
            column: 33
          },
          end: {
            line: 92,
            column: 41
          }
        }, {
          start: {
            line: 92,
            column: 44
          },
          end: {
            line: 92,
            column: 62
          }
        }],
        line: 92
      },
      "1": {
        loc: {
          start: {
            line: 96,
            column: 6
          },
          end: {
            line: 96,
            column: 53
          }
        },
        type: "cond-expr",
        locations: [{
          start: {
            line: 96,
            column: 16
          },
          end: {
            line: 96,
            column: 39
          }
        }, {
          start: {
            line: 96,
            column: 42
          },
          end: {
            line: 96,
            column: 53
          }
        }],
        line: 96
      }
    },
    s: {
      "0": 0,
      "1": 0,
      "2": 0,
      "3": 0,
      "4": 0,
      "5": 0,
      "6": 0,
      "7": 0,
      "8": 0,
      "9": 0,
      "10": 0,
      "11": 0,
      "12": 0,
      "13": 0,
      "14": 0
    },
    f: {
      "0": 0,
      "1": 0,
      "2": 0,
      "3": 0,
      "4": 0,
      "5": 0
    },
    b: {
      "0": [0, 0],
      "1": [0, 0]
    },
    _coverageSchema: "43e27e138ebf9cfc5966b082cf9a028302ed4184",
    hash: "00b494f305c68c531cf84782caae147fd4c42eac"
  };
  var coverage = global[gcv] || (global[gcv] = {});

  if (coverage[path] && coverage[path].hash === hash) {
    return coverage[path];
  }

  return coverage[path] = coverageData;
}();

import React, { useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import DisplayTrans from './DisplayTrans';
import { getTransactions } from '../../redux/actions/PlaidActions';
import { connect } from 'react-redux';
import './index.css';
import AddTransactionModal from '../Modal_Components/AddTransaction';
const useStyles = (cov_1x4u3utbdj.s[0]++, makeStyles(theme => {
  cov_1x4u3utbdj.f[0]++;
  cov_1x4u3utbdj.s[1]++;
  return {
    root: {
      '& > *': {
        margin: theme.spacing(1)
      }
    },
    container: {
      display: 'flex',
      width: '100%'
    },
    topcontent: {
      display: 'flex'
    },
    card: {
      width: '100%',
      marginBottom: '1rem',
      height: '70%',
      background: '#f7f7f7',
      textAlign: 'left',
      paddingLeft: '10px',
      display: 'flex',
      fontSize: '12px'
    },
    details: {
      display: 'flex'
    },
    content: {
      flex: '1 0 auto',
      alignItems: 'flex-start',
      textAlign: 'left'
    },
    controls: {
      display: 'flex',
      alignItems: 'center',
      textAlign: 'right',
      alignContent: 'flex-end'
    },
    text: {
      textAlign: 'left',
      paddingLeft: theme.spacing(5)
    },
    rightInfo: {
      width: '85%'
    },
    leftInfo: {
      width: '15%',
      alignItems: 'center'
    }
  };
}));
cov_1x4u3utbdj.s[2]++;

const Transactions = props => {
  cov_1x4u3utbdj.f[1]++;
  // This component displays the transactions table. 
  // filter determines the amount of transactions that are displayed. (True = 3 transactions, false = all transactions)
  // handleClick toggles the filter
  // open is used to display the add transaction modal
  const classes = (cov_1x4u3utbdj.s[3]++, useStyles());
  const [filter, setFilter] = (cov_1x4u3utbdj.s[4]++, useState(true));
  cov_1x4u3utbdj.s[5]++;

  const handleClick = e => {
    cov_1x4u3utbdj.f[2]++;
    cov_1x4u3utbdj.s[6]++;
    setFilter(!filter);
  };

  var selected = (cov_1x4u3utbdj.s[7]++, props.transactions.slice(0, 3));
  const [open, setOpen] = (cov_1x4u3utbdj.s[8]++, useState(false));
  cov_1x4u3utbdj.s[9]++;

  const handleClickOpen = () => {
    cov_1x4u3utbdj.f[3]++;
    cov_1x4u3utbdj.s[10]++;
    setOpen(true);
  };

  cov_1x4u3utbdj.s[11]++;

  const handleClose = () => {
    cov_1x4u3utbdj.f[4]++;
    cov_1x4u3utbdj.s[12]++;
    setOpen(false);
  };

  cov_1x4u3utbdj.s[13]++;
  return <React.Fragment>
			<div className='trans-container'>
				<div className='trans-top-content'>
					<h3 className={classes.text}>Recent Transactions:</h3>
						<button className='add-trans-button' onClick={handleClickOpen}>
							Add Transactions
						</button>
						<AddTransactionModal open={open} handleClose={handleClose} />
				</div>
				<div className='trans-item'>
					<DisplayTrans arr={filter ? (cov_1x4u3utbdj.b[0][0]++, selected) : (cov_1x4u3utbdj.b[0][1]++, props.transactions)} classes={classes} />
				</div>

				<button id='view-button' onClick={handleClick}>
					{filter ? (cov_1x4u3utbdj.b[1][0]++, "View all transactions") : (cov_1x4u3utbdj.b[1][1]++, "View less")}
				</button>
			</div>
	</React.Fragment>;
};

function mapStateToProps(state) {
  cov_1x4u3utbdj.f[5]++;
  cov_1x4u3utbdj.s[14]++;
  return {
    userID: state.loginReducer.user.id,
    LinkedAccount: state.loginReducer.user.LinkedAccount,
    transactions: state.plaidReducer.transactions
  };
}

export default connect(mapStateToProps, {
  getTransactions
})(Transactions);