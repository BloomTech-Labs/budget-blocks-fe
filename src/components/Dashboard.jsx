import React, { useEffect, useState } from 'react';
import { useOktaAuth } from '@okta/okta-react';
import { connect } from 'react-redux';
import { Link, Redirect } from 'react-router-dom';

import { Button } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';

// import { transitions } from 'react-alert';

// SECTION Redux Imports
import { fetchTransactions } from '../redux/actions/dashboardAction';
import { userAction, notAuthenticated } from '../redux/actions/userAction';

const Dashboard = ({
  userInfo,
  transaction,
  fetchTransactions,
  userAction,
  notAuthenticated,
  location,
  isFetching,
}) => {
  const { authState, authService } = useOktaAuth();

  const logout = async () => {
    localStorage.clear();
    authService.logout('/');
  };

  useEffect(() => {
    const { accessToken } = authState;

    if (!authState.isAuthenticated) {
      notAuthenticated();
    } else {
      authService.getUser().then((info) => {
        const oktaUserInfo = info;

        userAction(oktaUserInfo, accessToken);
      });
    }
  }, [authState, authService]);

  //FIXME FOR REFRESH BUTTON
  useEffect(() => {
    fetchTransactions();
  }, [location]);

  return (
    <div>
      {userInfo && userInfo.onboarding_complete === false ? (
        <Redirect to="/onboarding" />
      ) : (
        <div>
          <h1> DASHBOARD </h1>
          <p>
            If you landed here, then you have successfully logged in with Okta!!
          </p>
          <p
            style={{
              color: 'blue',
              fontSize: '32px',
            }}
          >
            Message:
            {`Hi, ${userInfo && userInfo.name}. Welcome to the dashboard.`}
          </p>
          <p> USER INFO STATE: {userInfo && userInfo.email} </p>
          <Link to="/onboarding">
            <Button color="primary" variant="contained">
              Onboarding
            </Button>
            <br />
            <br />
          </Link>
          <Button color="secondary" variant="contained" onClick={logout}>
            Logout
          </Button>
          <div style={{ textAlign: 'center' }}>
            {transaction.transactions &&
              transaction.transactions.map((data) => (
                <Card key={data.transaction_id}>
                  <CardContent>
                    <Typography style={{ fontSize: '3rem' }}>
                      {data.budget_blocks_category}
                    </Typography>
                    <Typography style={{ fontSize: '2rem' }}>
                      {data.amount}
                    </Typography>
                  </CardContent>
                </Card>
              ))}
          </div>
        </div>
      )}
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    transaction: state.trans.transaction,
    isFetching: state.trans.isFetching,
    errors: state.trans.errors,
    userInfo: state.users.userInfo,
  };
};

export default connect(mapStateToProps, {
  fetchTransactions,
  userAction,
  notAuthenticated,
})(Dashboard);
