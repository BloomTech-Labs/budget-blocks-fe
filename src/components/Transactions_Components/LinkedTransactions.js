import React, { useState} from "react";
import { makeStyles } from "@material-ui/core/styles";
import DisplayTrans from "./DisplayTrans";

import { getTransactions } from "../../redux/actions/PlaidActions";
import { connect } from "react-redux";
import "./index.css";

import LinkedTransaction from "../Modal_Components/LinkedTransaction";

const useStyles = makeStyles(theme => ({
  root: {
    "& > *": {
      margin: theme.spacing(1)
    }
  },
  container: {
    display: "flex",
    width: "100%"
  },
  topcontent: {
    display: "flex"
  },
  card: {
    width: "100%",
    marginBottom: "1rem",
    height: "70%",
    background: "#f7f7f7",
    textAlign: "left",
    paddingLeft: "10px",
    display: "flex",

    fontSize: "12px"
  },
  details: {
    display: "flex"
  },
  content: {
    flex: "1 0 auto",
    alignItems: "flex-start",
    textAlign: "left"
  },
  controls: {
    display: "flex",
    alignItems: "center",
    textAlign: "right",
    alignContent: "flex-end"
  },
  text: {
    textAlign: "left",
    paddingLeft: theme.spacing(5)
  },
  rightInfo: {
    width: "85%"
  },
  leftInfo: {
    width: "15%",
    alignItems: "center"
  }
}));

const Transactions = props => {
  // This component displays the transactions table.
  // filter determines the amount of transactions that are displayed. (True = 3 transactions, false = all transactions)
  // handleClick toggles the filter
  // open is used to display the add transaction modal
  const classes = useStyles();
  const [filter, setFilter] = useState(true);
  const handleClick = e => {
    setFilter(!filter);
  };



  var selected = props.transactions.slice(0, 3);
  const [open, setOpen] = useState(false);
  const handleClickOpen = () => {
    setOpen(true);
  };


  const handleClose = () => {
    setOpen(false);
  };

  return (
    <React.Fragment>
      <div className="trans-container">
        <div className="trans-top-content">
          <h3 className={classes.text}>Recent Transactions:</h3>
          <button className="add-trans-button" onClick={handleClickOpen}>
            Add Transactions
          </button>
          <LinkedTransaction open={open} handleClose={handleClose} />
        </div>
        <div className="trans-item">
          <DisplayTrans
            arr={filter ? selected : props.transactions}
            classes={classes}
          />
        </div>

        <button id="view-button" onClick={handleClick}>
          {filter ? "View all transactions" : "View less"}
        </button>
      </div>
    </React.Fragment>
  );
};
function mapStateToProps(state) {
  return {
    userID: state.loginReducer.user.id,
    LinkedAccount: state.loginReducer.user.LinkedAccount,
    transactions: state.plaidReducer.transactions
  };
}

export default connect(mapStateToProps, { getTransactions })(Transactions);
