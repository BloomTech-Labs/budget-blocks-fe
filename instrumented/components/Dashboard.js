var cov_1ftat482c3 = function () {
  var path = "/Users/Cha/Desktop/Lambda/budget-blocks-fe/src/components/Dashboard.js";
  var hash = "4f1460de00e07f6b507b60a8c71704e9dc5b72a8";
  var global = new Function("return this")();
  var gcv = "__coverage__";
  var coverageData = {
    path: "/Users/Cha/Desktop/Lambda/budget-blocks-fe/src/components/Dashboard.js",
    statementMap: {
      "0": {
        start: {
          line: 14,
          column: 25
        },
        end: {
          line: 35,
          column: 1
        }
      },
      "1": {
        start: {
          line: 16,
          column: 1
        },
        end: {
          line: 20,
          column: 27
        }
      },
      "2": {
        start: {
          line: 17,
          column: 8
        },
        end: {
          line: 19,
          column: 63
        }
      },
      "3": {
        start: {
          line: 21,
          column: 1
        },
        end: {
          line: 34,
          column: 3
        }
      },
      "4": {
        start: {
          line: 38,
          column: 1
        },
        end: {
          line: 44,
          column: 3
        }
      }
    },
    fnMap: {
      "0": {
        name: "(anonymous_0)",
        decl: {
          start: {
            line: 14,
            column: 25
          },
          end: {
            line: 14,
            column: 26
          }
        },
        loc: {
          start: {
            line: 14,
            column: 34
          },
          end: {
            line: 35,
            column: 1
          }
        },
        line: 14
      },
      "1": {
        name: "(anonymous_1)",
        decl: {
          start: {
            line: 16,
            column: 11
          },
          end: {
            line: 16,
            column: 12
          }
        },
        loc: {
          start: {
            line: 16,
            column: 17
          },
          end: {
            line: 20,
            column: 2
          }
        },
        line: 16
      },
      "2": {
        name: "mapStateToProps",
        decl: {
          start: {
            line: 37,
            column: 9
          },
          end: {
            line: 37,
            column: 24
          }
        },
        loc: {
          start: {
            line: 37,
            column: 32
          },
          end: {
            line: 45,
            column: 1
          }
        },
        line: 37
      }
    },
    branchMap: {
      "0": {
        loc: {
          start: {
            line: 17,
            column: 8
          },
          end: {
            line: 19,
            column: 62
          }
        },
        type: "cond-expr",
        locations: [{
          start: {
            line: 18,
            column: 14
          },
          end: {
            line: 18,
            column: 49
          }
        }, {
          start: {
            line: 19,
            column: 14
          },
          end: {
            line: 19,
            column: 62
          }
        }],
        line: 17
      }
    },
    s: {
      "0": 0,
      "1": 0,
      "2": 0,
      "3": 0,
      "4": 0
    },
    f: {
      "0": 0,
      "1": 0,
      "2": 0
    },
    b: {
      "0": [0, 0]
    },
    _coverageSchema: "43e27e138ebf9cfc5966b082cf9a028302ed4184",
    hash: "4f1460de00e07f6b507b60a8c71704e9dc5b72a8"
  };
  var coverage = global[gcv] || (global[gcv] = {});

  if (coverage[path] && coverage[path].hash === hash) {
    return coverage[path];
  }

  return coverage[path] = coverageData;
}();

import React, { useEffect } from 'react';
import LinkedBlocks from './Blocks_Components/LinkedBlocks';
import { connect } from 'react-redux';
import LinkedTransactions from './Transactions_Components/LinkedTransactions';
import TotalBudget from './TotalBudget_Components/TotalBudget';
import { getTransactions } from '../redux/actions/PlaidActions';
import './dashboardStyle.css';
import 'react-loader-spinner/dist/loader/css/react-spinner-loader.css';
import '../App.css';
import './main.css';
import { getManualTrans } from "../redux/actions/ManualActions";
import Loading from "./Loading";
cov_1ftat482c3.s[0]++;
export const Dashboard = props => {
  cov_1ftat482c3.f[0]++;
  cov_1ftat482c3.s[1]++;
  // This component displays the dashboard: Blocks, transactions, and budget
  useEffect(() => {
    cov_1ftat482c3.f[1]++;
    cov_1ftat482c3.s[2]++;
    props.LinkedAccount == true ? (cov_1ftat482c3.b[0][0]++, props.getTransactions(props.userID)) : (cov_1ftat482c3.b[0][1]++, props.getManualTrans(props.userID, props.history));
  }, [props.LinkedAccount]);
  cov_1ftat482c3.s[3]++;
  return <div className='app-container'>
			<Loading />
			<div className='showcase'>
				<div className='right-showcase'>
						<TotalBudget />
				</div>
				<div className='left-showcase'>
						<LinkedBlocks /> 
						<LinkedTransactions />		
				</div>
			</div>
		</div>;
};

function mapStateToProps(state) {
  cov_1ftat482c3.f[2]++;
  cov_1ftat482c3.s[4]++;
  return {
    userID: state.loginReducer.user.id,
    LinkedAccount: state.loginReducer.user.LinkedAccount,
    blocks: state.plaidReducer.categories,
    plaidFetching: state.plaidReducer.isFetching,
    blockFetching: state.blockReducer.isFetching
  };
}

export default connect(mapStateToProps, {
  getTransactions,
  getManualTrans
})(Dashboard);