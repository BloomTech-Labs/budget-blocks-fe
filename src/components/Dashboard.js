import React, { useEffect } from "react";
import LinkedBlocks from "./Blocks_Components/LinkedBlocks";
import { connect } from "react-redux";
import LinkedTransactions from "./Transactions_Components/LinkedTransactions";
import TotalBudget from "./TotalBudget_Components/TotalBudget";
import { getTransactions } from "../redux/actions/PlaidActions";
import "./dashboardStyle.css";
import "react-loader-spinner/dist/loader/css/react-spinner-loader.css";
import "../App.css";
import "./main.css";
import { getManualTrans } from "../redux/actions/ManualActions";
import Loading from "./Loading";
import { PageView } from "./google_analytics/index.js";
import { updateExpense, setBlockHover,selectExpense, getExpenses, deleteExpense, addExpense, assignBlock, unassignExpense } from '../redux/actions/ExpenseActions'
import { getBlocks, deleteBlock, updateBlock, addBlock, addOwnExpense, deleteAndSave } from '../redux/actions/BlockActions'

import ExpenseList from './no-bank/ExpenseList'

export const Dashboard = (props) => {
  // This component displays the dashboard: Blocks, transactions

  useEffect(() => {
    // get all of a user's expeses
    // pass expenses to expenseList
    props.getExpenses(props.userID);
    props.getBlocks(props.userID);
  }, []);

  useEffect(() => {
    PageView();
  }, []);

  return (
    <div className="app-container">
      <Loading />
      <div className="showcase">
        <ExpenseList
          userID={props.userID}
          handleDeleteExpense={props.deleteExpense}
          expenses={props.expenses}
          handleAddExpense={props.addExpense}
          handleAddBlock={props.addBlock}
          blocks={props.blocks}
          handleAssignBlock={props.assignBlock}
          handleAddOwnExpense={props.addOwnExpense}
          blockExpenses={props.blockExpenses}
          handleUnassignExpense={props.unassignExpense}
          handleDeleteAndSave={props.deleteAndSave}
          handleDeleteBlock={props.deleteBlock}
          handleUpdateBlock={props.updateBlock}
          handleUpdateExpense={props.updateExpense}
          handleSelectExpense={props.selectExpense}
          selectedExpense={props.selectedExpense}
          blockHover={props.blockHover}
          handleSetBlockHover={props.setBlockHover}
        />
      </div>
    </div>
  );
};

function mapStateToProps(state) {
  return {
    userID: state.loginReducer.user.id,
    expenses: state.expenses.expenses,
    blocks: state.blocks.blocks,
    blockExpenses: state.blocks.blockExpenses,
    selectedExpense: state.expenses.selectedExpense,
    blockHover: state.expenses.blockHover
  };
}

export default connect(mapStateToProps, {
  getExpenses,
  deleteExpense,
  addExpense,
  getBlocks,
  addBlock,
  assignBlock,
  addOwnExpense,
  unassignExpense,
  deleteAndSave,
  deleteBlock,
  updateBlock,
  updateExpense,
  selectExpense,
  setBlockHover
})(
  Dashboard
);
