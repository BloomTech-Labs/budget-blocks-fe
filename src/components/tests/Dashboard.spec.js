import React from "react";
import { Dashboard } from "../Dashboard";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import LinkedTransactions from "../Dashboard";

configure({ adapter: new Adapter() });

test("Dashboard renders correctly", () => {
  const wrapper = shallow(<Dashboard LinkedAccount={false} blocks={[]} />);

  expect(wrapper.exists()).toBe(true);
});
