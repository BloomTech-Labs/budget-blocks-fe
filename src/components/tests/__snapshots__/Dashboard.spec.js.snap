// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders correctly 1`] = `ShallowWrapper {}`;
